src/ray/core_worker/core_worker.h

```diff
@@ -1873,6 +1877,14 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   uint32_t pid_;
 
   absl::flat_hash_set<ObjectID> deleted_generator_ids_;
+
+  /// TODO(hjiang): Cached job runtime env info.
+  ///
+  /// Serialized runtime info env are cached.
+  mutable std::mutex job_runtime_env_serialization_mutex_;
```

Comment:
we can use ThreadSafeSharedLruConstCache to avoid using this mutex here?

Reply:
No it's intentionally, using mutex guarantees for one particular key (aka, serialized runtime info), only one parsing will happen, so we don't have repeated operation.

Reply:
To reduce lock contention, I'm planning to implement a sharded LRU with a `GetOrCalculate`-like interface, TODO already left in the shared lru file.

Reply:
then I guess we can remove ThreadSafeSharedLruConstCache lol.